src/reader/index.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const CID = require('cids')
+const pull = require('pull-stream')
+const asyncValues = require('pull-async-values')
+const asyncMap = require('pull-stream/throughs/async-map')
+const map = require('pull-stream/throughs/map')
+const UnixFS = require('ipfs-unixfs')
+const toB58String = require('multihashes').toB58String
+const waterfall = require('async/waterfall')
+
+module.exports = (path, ipldResolver, begin = 0, end) => {
+  let streamPosition = 0
+
+  return pull(
+    asyncValues((cb) => {
```

Comment:
Interesting, didn't know about pull-async-values.
Couldn't this be done instead with something like a `pull.value([path])` and then using a chain of `pull.map` and `pull.asyncMap`?
I'm asking because this way we wouldn't need to add an extra dependency..

Reply:
I couldn't find an easy way to asynchronously supply an array of values to `pull-stream` when you don't know the length of the values ahead of time using only the built-in functions.

The faq recommends using external modules for [1:many transforms](https://github.com/pull-stream/pull-stream-faq#how-can-i-achieve-1many-and-many1-transforms) and [async steps](https://github.com/pull-stream/pull-stream-faq#i-need-to-return-a-stream-immediately-but-need-to-do-something-async-first-how) so I'm not sure it's possible without adding an extra dep.

That said, the source for `pull-async-values` is pretty small - I can pull it in to this repo if required.

Reply:
@achingbrain Thanks for explaining, it's fine as it is by me :)